### PR TITLE
Fix golang check so that it performs arithemtic comparison

### PR DESCRIPTION
**Pull request by soltysh**
/assign @stevekuznetsov

### CODE REVIEW

**Comment by tnozicka** [comment]
https://github.com/kubernetes/kubernetes/pull/58207/files

**Comment by soltysh** [comment]
Upstream uses different scripts than we do, and we already have that awk in place when checking glide. See the same file several lines below ;)

**Comment by stevekuznetsov** [comment]
This is because of the sorting issues around `1.10`? OK.

**Comment by stevekuznetsov** [comment]
Crazy but it just might work
/lgtm
/retest

**Comment by soltysh** [comment]
/hold
I need to verify this, it's retesting too frequently.

File: hack/lib/util/golang.sh [hunk 1]
```diff
@@ -8,7 +8,7 @@ function os::golang::verify_go_version() {
 
 	local go_version
 	go_version=($(go version))
-	if [[ "${go_version[2]}" < "${OS_REQUIRED_GO_VERSION}" ]]; then
+	if ! echo "${go_version[2]#go}" | awk -F. -v min=${OS_REQUIRED_GO_VERSION#go} '{ exit $2 < min }'; then
```
Comment (cben) [inline hack/lib/util/golang.sh]: AFAICT this logic is badly broken, e.g. it allows go1.2 but will fail for go2.0.
`-F.` means `$2` is just the second (minor) component, e.g. for go1.11.4 it's `11`.
`$OS_REQUIRED_GO_VERSION` looks like `go1.10`.
(see https://github.com/openshift/origin/blob/v4.0.0-alpha.0/hack/lib/constants.sh#L7, and  #19042 backport used hardcoded `min=1.8`) 
So awk is comparing a *minor* integer like 11 < *major.minor* float like 1.9!
Anything with minor version 2 or higher will pass, while completely ignoring go's major version!
```
$ go_version=(go version go1.0.0 linux/amd64); echo  "${go_version[2]#go}" | awk -F. -v min=1.8 '{ print $2, "<", min, "=", $2 < min }'
0 < 1.8 = 1
$ go_version=(go version go1.1.1 linux/amd64); echo  "${go_version[2]#go}" | awk -F. -v min=1.8 '{ print $2, "<", min, "=", $2 < min }'
1 < 1.8 = 1
$ go_version=(go version go1.2.0 linux/amd64); echo  "${go_version[2]#go}" | awk -F. -v min=1.8 '{ print $2, "<", min, "=", $2 < min }'
2 < 1.8 = 0
$ go_version=(go version go0.2.0 linux/amd64); echo  "${go_version[2]#go}" | awk -F. -v min=1.8 '{ print $2, "<", min, "=", $2 < min }'
2 < 1.8 = 0
$ go_version=(go version go99.1.99 linux/amd64); echo  "${go_version[2]#go}" | awk -F. -v min=1.8 '{ print $2, "<", min, "=", $2 < min }'
1 < 1.8 = 1
```

https://github.com/kubernetes/kubernetes/pull/58207 tuple comparison looks better (probably would need trimming to 2 components)

Comment (soltysh) [reply hack/lib/util/golang.sh]: @cben feel free to open a PR fixing this and ping me there :)

**Pull request merged**